langchain4j-alloydb-pg/src/main/java/dev/langchain4j/data/document/loader/alloydb/AlloyDBLoader.java

```diff
@@ -35,95 +35,140 @@ public class AlloyDBLoader {
+                case "YAML":
+                    this.formatter = AlloyDBLoader::yamlFormatter;
+                    break;
+                default:
+                    this.formatter = AlloyDBLoader::textFormatter;
             }
+            return this;
         }
 
-        if (metadataJsonColumn != null && !columnNames.contains(metadataJsonColumn)) {
-            throw new IllegalArgumentException(String.format("Column %s not found in query result %s.", metadataJsonColumn, columnNames));
+        public Builder contentColumns(List<String> contentColumns) {
+            this.contentColumns = contentColumns;
+            return this;
         }
 
-        if (metadataJsonColumn == null && columnNames.contains(DEFAULT_METADATA_COL)) {
-            metadataJsonColumn = DEFAULT_METADATA_COL;
-        } else {
-            metadataJsonColumn = null;
+        public Builder metadataColumns(List<String> metadataColumns) {
+            this.metadataColumns = metadataColumns;
+            return this;
         }
 
-        List<String> allNames = new ArrayList<>(contentColumns);
-        allNames.addAll(metadataColumns);
-        if (metadataJsonColumn != null) {
-            allNames.add(metadataJsonColumn);
+        public Builder metadataJsonColumn(String metadataJsonColumn) {
+            this.metadataJsonColumn = metadataJsonColumn;
+            return this;
         }
 
-        for (String name : allNames) {
-            if (!columnNames.contains(name)) {
-                throw new IllegalArgumentException(String.format("Column %s not found in query result %s.", name, columnNames));
+        public AlloyDBLoader build() throws SQLException {
+            if ((this.query == null || this.query.isEmpty()) && (this.tableName == null || this.tableName.isEmpty())) {
+                throw new IllegalArgumentException("Either query or tableName must be specified.");
+            }
+            if (query == null) {
+                query = String.format("SELECT * FROM \"%s\".\"%s\" LIMIT 1", schemaName, tableName);
+            }
+
+            List<String> columnNames = new ArrayList<>();
+            try (Connection pool = engine.getConnection(); PreparedStatement statement = pool.prepareStatement(query); ResultSet resultSet = statement.executeQuery()) {
+                this.connection = pool;
+                for (int i = 1; i <= resultSet.getMetaData().getColumnCount(); i++) {
+                    columnNames.add(resultSet.getMetaData().getColumnName(i));
+                }
+            }
+
+            contentColumns = contentColumns == null || contentColumns.isEmpty() ? List.of(columnNames.get(0)) : contentColumns;
+            metadataColumns = metadataColumns == null ? new ArrayList<>() : metadataColumns;
+            for (String col : columnNames) {
+                if (!contentColumns.contains(col) && !metadataColumns.contains(col)) {
+                    metadataColumns.add(col);
+                }
             }
```

Comment:
This logic should be either use the set metadataColumns or use the all columns not included in content columns list. This however, will append more metadata columns if even metadatacolumns is found.